### ISSUE TITLE
Global phase not preserved by transpiler

### ISSUE THREAD

**Issue by juanjosegarciaripoll** [Qiskit/qiskit-terra#5950]
### Information

- **Qiskit Terra version**: 0.16.4
- **Python version**: 3.9.1
- **Operating system**: Windows

### What is the current behavior?

The global phase in a wavefunction generated by statevector-simulator depends on the optimization level in the transpiler.

### Steps to reproduce the problem

```
>>> from qiskit import QuantumRegister, QuantumCircuit, BasicAer, execute
>>> from qiskit.circuit.library import RYGate
>>> import numpy as np
>>> register = QuantumRegister(1)
>>> backend = BasicAer.get_backend('statevector_simulator')
>>> circuit = QuantumCircuit(register)
>>> circuit.x(0)
<qiskit.circuit.instructionset.InstructionSet object at 0x0000024AA8E4EB20>
>>> ψ1 = execute(circuit, backend).result().get_statevector()
>>> print('One state |1> wavefunction ψ1=\n', ψ1)
One state |1> wavefunction ψ1=
 [0.+0.j 1.+0.j]
>>> theta = np.pi/3.0
>>> U = RYGate(theta).to_matrix()
>>> print('Qiskit U=\n', U)
Qiskit U=
 [[ 0.8660254+0.j -0.5      +0.j]
 [ 0.5      +0.j  0.8660254+0.j]]
>>> circuit = QuantumCircuit(register)
>>> circuit.x(0)
<qiskit.circuit.instructionset.InstructionSet object at 0x0000024AA8DE9520>
>>> circuit.ry(theta, 0)
<qiskit.circuit.instructionset.InstructionSet object at 0x0000024AA8E4EB20>
>>> Uψ1 = execute(circuit, backend).result().get_statevector()
>>> print('Obtained U * |1> =\n', Uψ1)
Obtained U * |1> =
 [ 0.5      +0.00000000e+00j -0.8660254+1.06057524e-16j]
>>> print('Expected U * |1> =\n', U @ ψ1)
Expected U * |1> =
 [-0.5      +0.j  0.8660254+0.j]
>>> circuit = QuantumCircuit(register)
>>> circuit.x(0)
<qiskit.circuit.instructionset.InstructionSet object at 0x0000024AA8E4FA30>
>>> circuit.ry(theta, 0)
<qiskit.circuit.instructionset.InstructionSet object at 0x0000024AA8E4EB20>
>>> Uψ1 = execute(circuit, backend, optimization_level=0).result().get_statevector()
>>> print('Obtained U * |1> =\n', Uψ1)
Obtained U * |1> =
 [-0.5      +0.j  0.8660254+0.j]
>>> print('Expected U * |1> =\n', U @ ψ1)
Expected U * |1> =
 [-0.5      +0.j  0.8660254+0.j]
```

### What is the expected behavior?

The output should not depend on optimization_level

**Comment by mtreinish** [comment]
So I actually think this is a bug with BasicAer not factoring in a global phase we set on the output circuit in the transpiler, and not a bug in the transpiler itself. When I ran this example using Aer:

```python3
from qiskit import *
from qiskit.circuit.library import RYGate
import numpy as np

backend = Aer.get_backend('statevector_simulator')
theta = np.pi / 3.0
circuit = QuantumCircuit(1)
circuit.x(0)
circuit.ry(theta, 0)

U_psi_0 = execute(circuit, backend, optimization_level=0).result().get_statevector()
print("Optimization_level 0:")
print(U_psi_0)
U_psi_1 = execute(circuit, backend).result().get_statevector()
print("Optimization_level 1:")
print(U_psi_1)
U_psi_2 = execute(circuit, backend, optimization_level=2).result().get_statevector()
print("Optimization_level 1:")
print(U_psi_2)
U_psi_3 = execute(circuit, backend, optimization_level=3).result().get_statevector()
print("Optimization_level 3:")
print(U_psi_3)
```

it produces the expected result (ignoring float precision):

```
Optimization_level 0:
[-0.5      +0.j  0.8660254+0.j]
Optimization_level 1:
[-0.5      -6.123234e-17j  0.8660254+0.000000e+00j]
Optimization_level 1:
[-0.5      -6.123234e-17j  0.8660254+0.000000e+00j]
Optimization_level 3:
[-0.5      -6.123234e-17j  0.8660254+0.000000e+00j]
```

**Comment by ewinston** [comment]
@juanjosegarciaripoll  Has this issue been resolved? I don't see it on BasicAer as of `0.18.0.dev0+31a24c1`

**Comment by ewinston** [comment]
Closing this issue. If it still persists please feel free to reopen.
